### PR TITLE
feat: implement Hadolint DL4005 — use SHELL to change the default shell

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Port Hadolint DL4005: warns when `RUN` uses `ln` to symlink `/bin/sh` instead of the `SHELL` instruction
- Auto-fix (`FixSuggestion`) replaces `ln /bin/sh` with `SHELL ["/bin/bash", "-c"]`, handling all `&&` chain positions (standalone, start, middle, end)
- New generic `shell.FindCommandInChain` utility for locating and extracting a command from a `&&` pipeline — reusable by future rules

## Test plan

- [x] All 3 original Hadolint DL4005Spec test cases ported and passing
- [x] Additional edge cases: exec form, multi-stage, chained commands
- [x] Fix tests for all chain positions (standalone, start, middle, end, exec form no-fix)
- [x] `shell.FindCommandInChain` unit tests (standalone, start, middle, end, no-match, predicate filtering)
- [x] Integration tests: check + fix snapshots
- [x] Full test suite passes (`go test ./internal/...`)
- [x] Lint passes (`make lint`)
- [x] `hadolint-status.json`, `RULES.md`, `README.md` updated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added DL4005 lint rule to detect ln -> /bin/sh patterns and recommend/apply SHELL-based fixes.
* **Tests**
  * Added unit and integration tests exercising DL4005 detection and fix behaviors.
* **Documentation**
  * Updated README and rules summary to include the new rule and counts.
* **Chores**
  * Updated snapshots and rule tallies (rules enabled increased by 1).

### CODE REVIEW

File: internal/shell/chain_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	if pos == nil {
+		t.Fatal("expected to find command")
+	}
+	if pos.PrecedingCommands != "apt-get update" {
+		t.Errorf("PrecedingCommands = %q, want %q", pos.PrecedingCommands, "apt-get update")
+	}
+	if pos.RemainingCommands != "echo done" {
+		t.Errorf("RemainingCommands = %q, want %q", pos.RemainingCommands, "echo done")
+	}
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/shell/chain_test.go]: ✅ Addressed in b577ebd. Added tests for semicolon separators, pipe operators, and mixed separator patterns. Thanks @coderabbitai\!
